components/region_cache_memory_engine/src/read.rs

```diff
@@ -417,6 +435,14 @@ impl Iterator for RangeCacheIterator {
         if self.valid {
             self.find_next_visible_key(true, guard);
         }
+
+        if self.valid {
```

Comment:
why not merge with the self.valid above.

Reply:
Also the code is duplicated by many times, make it an shared inline function

Reply:
after find_next_visible_key, the self.valid may not be true.

Reply:
Please add a comment to mention that tself.valid can be changed. It's not straightforward.

Reply:
But we should not have duplicated code when possible

Reply:
I have polished that.